sys/kern/src/arch/rv64/mtimer.rs

```diff
@@ -34,18 +32,8 @@ pub unsafe fn set_timer(tick_divisor: u32) {
     // Set high-order bits of mtime to zero.  We only call this function prior
     // to enabling interrupts so it should be safe.
     unsafe {
-        asm!("
```

Comment:
From comment above, "// On both RV32 and RV64 systems the mtime and mtimecmp memory-mapped registers
// are 64-bits wide."

That's not architecturally true in the generic sense. It's implementation defined.

Reply:
Yeah, this file needs more work as it also makes assumptions about the addresses for mtime and mtimecmp. [SW-606](http://go/ji/SW-606) tracks this.